barretenberg/cpp/src/barretenberg/flavor/flavor.hpp

```diff
@@ -214,14 +215,35 @@ template <typename PrecomputedCommitments> class NativeVerificationKey_ : public
     // transcript produced hash.
     fr hash()
     {
-        fr challenge = crypto::Poseidon2<crypto::Poseidon2Bn254ScalarFieldParams>::hash(to_field_elements());
+        fr challenge = crypto::Poseidon2<crypto::Poseidon2Bn254ScalarFieldParams>::hash(this->to_field_elements());
         // match the parameter used in stdlib, which is derived from cycle_scalar (is 128)
         static constexpr size_t LO_BITS = fr::Params::MAX_BITS_PER_ENDOMORPHISM_SCALAR;
 
         auto converted = static_cast<uint256_t>(challenge);
         uint256_t lo = converted.slice(0, LO_BITS);
         return lo;
     }
+
+    /**
+     * @brief Adds the verification key witnesses directly to the transcript.
+     * @details Only needed to make sure the Origin Tag system works. Rather than converting into a vector of fields and
```

Comment:
Can you reword this slightly to make it clear that we need VK hashing for soundness but that we do it in this particular way to satisfy the Origin Tag system? Right now it almost reads like this is strictly a debugging/tooling method